src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -453,10 +483,29 @@ def get_temperature(actual_temperature_lines: list[str] | None) -> float | None:
     )
 
 
+def get_kinetic_measurements(
+    kinetic_lines: list[str] | None,
+) -> tuple[dict[str, list[float | None]], list[float]] | None:
+    if not kinetic_lines:
+        return None
+    data = read_csv(StringIO("\n".join(kinetic_lines)), sep="\t")
+    kinetic_measurements: defaultdict[str, list[float | None]] = defaultdict(
+        list[float | None]
+    )
+    kinetic_elapsed_time: list[float] = []
+    for col in data.columns:
+        for _idx, value in enumerate(data[col]):
```

Comment:
With regard to this iteration, I think you can simplify this a bit using `pandas` `to_list()`:

```
    kinetic_elapsed_time = data["Time"].map(lambda val: _convert_time_to_seconds(str(val))).to_list()
    for col_name, column in data.items():
        if col_name == "Time":
            continue
        if column.isnull().any():
            msg = f"Unable to process null value in column: {col_name} of Kinetic section."
            raise AllotropeConversionError(msg)
        kinetic_measurements[str(col_name)] = column.astype(float).to_list()
    return dict(kinetic_measurements.items()), kinetic_elapsed_time
```